src/main/java/TaskList.java

```diff
@@ -0,0 +1,196 @@
+            if (storedTaskString.startsWith("[T]")) {
+                String task = getTodoTask(storedTaskString);
+
+                createTodoTask(task);
+                checkIfTaskDone(storedTaskString);
+            } else if (storedTaskString.startsWith("[D]")) {
+                String task = getDeadlineTask(storedTaskString);
+                String date = getDeadlineDate(storedTaskString);
+
+                DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM d yyyy");
+                LocalDate by = LocalDate.parse(date, format);
+
+                createDeadlineTask(task, by);
+                checkIfTaskDone(storedTaskString);
+            } else if (storedTaskString.startsWith("[E]")) {
+                String task = getEventTask(storedTaskString);
+                String at = getEventDate(storedTaskString);
+
+                createEventTask(task, at);
+                checkIfTaskDone(storedTaskString);
+            } else {
+                throw new DukeException("Invalid task.");
+            }
+        }
+    }
+
+    /**
+     * Returns todo task description from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the todo task description
+     */
+    public static String getTodoTask(String input) {
+        return input.substring(16);
+    }
+
+    /**
+     * Returns deadline task description from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the deadline task description
+     */
+    public static String getDeadlineTask(String input) {
+        int dateIndex = input.indexOf("(by: ");
+        return input.substring(16, dateIndex - 1);
+    }
+
+    /**
+     * Returns deadline task date from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the deadline task date
+     */
+    public static String getDeadlineDate(String input) {
+        int dateIndex = input.indexOf("(by: ");
+        return input.substring(dateIndex + 5, dateIndex + 15);
+    }
+
+    /**
+     * Returns event task description from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the event task description
+     */
+    public static String getEventTask(String input) {
+        int dateIndex = input.indexOf("(at: ");
+        return input.substring(16, dateIndex - 1);
+    }
+
+    /**
+     * Returns event task date from filtering the input
+     * @param input the task line previously loaded into the file
+     * @return the event task date
+     */
+    public static String getEventDate(String input) {
+        int dateIndex = input.indexOf("(at: ");
+        return input.substring(dateIndex + 5, input.length() - 2);
+    }
+
+    /**
+     * Checks if the task stored in the file has been completed
+     * @param storedTaskString task stored in the file
+     */
+    public static void checkIfTaskDone(String storedTaskString) {
```

Comment:
Perhaps you could consider a different method name here since checkifTaskDone sounds like a boolean method, however, it performs stateful operations